sdk/iot/azure-iot-modelsrepository/tests/test_dtmi_conventions.py

```diff
@@ -0,0 +1,139 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import pytest
+from azure.iot.modelsrepository import dtmi_conventions
+
+
+@pytest.mark.describe(".is_valid_dtmi()")
+class TestIsValidDTMI(object):
+    @pytest.mark.it("Returns True if given a valid DTMI")
+    @pytest.mark.parametrize(
+        "dtmi",
+        [
+            pytest.param("dtmi:FooDTDL;1", id="Short DTMI"),
+            pytest.param("dtmi:com:somedomain:example:FooDTDL;1", id="Long DTMI"),
+        ],
+    )
+    def test_valid_dtmi(self, dtmi):
+        assert dtmi_conventions.is_valid_dtmi(dtmi)
+
+    @pytest.mark.it("Returns False if given an invalid DTMI")
+    @pytest.mark.parametrize(
+        "dtmi",
+        [
+            pytest.param("", id="Empty string"),
+            pytest.param("not a dtmi", id="Not a DTMI"),
+            pytest.param("com:somedomain:example:FooDTDL;1", id="DTMI missing scheme"),
+            pytest.param("dtmi:com:somedomain:example:FooDTDL", id="DTMI missing version"),
+            pytest.param("dtmi:foo_bar:_16:baz33:qux;12", id="System DTMI"),
+        ],
+    )
+    def test_invalid_dtmi(self, dtmi):
+        assert not dtmi_conventions.is_valid_dtmi(dtmi)
+
+
+@pytest.mark.describe(".get_model_uri()")
+class TestGetModelURI(object):
+    @pytest.mark.it("Returns the URI for a specified model at a specified repository")
+    @pytest.mark.parametrize(
+        "dtmi, repository_uri, expected_model_uri",
+        [
+            pytest.param(
+                "dtmi:com:somedomain:example:FooDTDL;1",
+                "https://myrepository/",
+                "https://myrepository/dtmi/com/somedomain/example/foodtdl-1.json",
+                id="HTTPS repository URI",
+            ),
+            pytest.param(
+                "dtmi:com:somedomain:example:FooDTDL;1",
+                "http://myrepository/",
+                "http://myrepository/dtmi/com/somedomain/example/foodtdl-1.json",
+                id="HTTP repository URI",
+            ),
+            pytest.param(
+                "dtmi:com:somedomain:example:FooDTDL;1",
+                "file:///myrepository/",
+                "file:///myrepository/dtmi/com/somedomain/example/foodtdl-1.json",
+                id="Filesystem URI",
+            ),
+            pytest.param(
```

Comment:
Would be good to add a test case for basic windows UNC path...as an example getting the model uri for "dtmi:com:example:Thermostat;1" in the repository location of "\\\\server\\\\repository" should net the uri of "file://server/repository/dtmi/com/example/thermostat-1.json"

Reply:
The function `.get_model_uri()` has been scoped to require the base path to already be provided in URI format (as opposed to the ModelsRepositoryClient itself, which does support non-URI filepaths). If you wish to expand it's functionality to support absolute paths being converted to URI, this would be an additional feature that can be added at a later date.